src/execution_tree/primitives/base_primitive.cpp

```diff
@@ -656,7 +687,8 @@ namespace phylanx { namespace execution_tree
         {
         case 0: HPX_FALLTHROUGH;    // nil
         case 1: HPX_FALLTHROUGH;    // phylanx::ir::node_data<std::uint8_t>
-        case 2: HPX_FALLTHROUGH;    // std::uint64_t
+        case 2:                     // ir::node_data<std::int64_t>
+            return std::move(val);
```

Comment:
This change would make `case 0` and `case 1` fall through to `case 2` which is not what we want.